library/core/src/ops/deref.rs

```diff
@@ -275,6 +275,17 @@ impl<T: ?Sized> DerefMut for &mut T {
     }
 }
 
+/// UwU
+#[unstable(feature = "deref_pure_trait", issue = "87121")]
```

Comment:
What's your thinking for making this a new feature gate?

Reply:
Sorry, should have explained. The feature is marked as incomplete and we don't allow using incomplete features in libcore.

Reply:
Oh I see. Just checking: I'd expect this to apply to uses of a feature, i.e. `#![feature(foo)]` in `core/lib.rs`, but this isn't the case here. Does that requirement also apply when we're only using the feature as a glorified `cfg_attr`?

Reply:
Uses of a feature include implementation of a trait gated by that feature. Since we have `unsafe impl DerefPure` in libcore/liballoc, we're technically using the feature, which triggers the `incomplete_features` lint.

Reply:
Ok, fair. Also I just realized that when we stabilize deref patterns we will very likely not want to stabilize that trait at the same time so it actually makes a lot of sense to have them separate

Reply:
That is a good point that I totally overlooked 👍